bundler/lib/bundler/source/rubygems.rb

```diff
@@ -172,7 +172,7 @@ def install(spec, options = {})
           :bin_dir => bin_path.to_s,
           :ignore_dependencies => true,
           :wrappers => true,
-          :env_shebang => true,
+          :env_shebang => Bundler.settings[:shebang],
```

Comment:
:thinking: I see there was some discussion for this at #6478 already. Looking at this again, `Bundler.settings[:shebang]` is `nil` by default. To keep this backward compatible, would it make sense to keep the default `true`, but respect `Bundler.settings[:shebang]` when set (aka when not `nil`)? Also some test would be great to cover this.